app/Http/Controllers/VisitPunchController.php

```diff
@@ -103,8 +114,16 @@ public function store(StoreVisitPunch $request)
 
             // Check if kiosk/door where punched is part of a different space
             $punch_space_id = (int) $request->input('space_id');
+            $punch_space = Space::find($punch_space_id);
             $active_visit_space_ids = $active_visits->first()->spaces->pluck('id')->toArray();
-            if (in_array($punch_space_id, $active_visit_space_ids, true)) {
+            $punch_space_is_parent = false;
+            foreach ($active_visit_space_ids as $active_visit_space_id) {
+                if ($punch_space->children->contains($active_visit_space_id)) {
+                    $punch_space_is_parent = true;
+                    continue;
```

Comment:
```suggestion
                    break;
```

Reply:
Ah drat that's what I meant to do, thanks.